apps/sms/js/activity_handler.js

```diff
@@ -229,25 +229,24 @@ var ActivityHandler = {
     //Validate if message still exists before opening message thread
     //See issue https://bugzilla.mozilla.org/show_bug.cgi?id=837029
     if (!message) {
-      return;
+      return Promise.reject(new Error('Message is not defined!'));
```

Comment:
> I think I'd simply return a Promise.resolve(). Returning a rejected promise is just like throwing an error.

But if message is not defined it's definitely something not expected, isn't it? I see that we have only two places where we call this method: on sms received - if message is undefined there, then we'll get exception before we call this one, and the second one is when we handle notification click where ```message``` is always defined
```
ActivityHandler.handleMessageNotification({
  id: params.id,
  threadId: params.threadId
});
```

So maybe we can just get rid of this ```if``` - I don't really think it's useful at all?